(GH-2361) Deprecate puppetfile commands and config option (#2457)
puppetlabs/bolt: documentation/modules.md

You can learn more about the module changes introduced in Bolt 2.23.0 in the
[September 2020 developer update](./developer_updates.md#september-2020).

> ## Migrate a project to use module management

Reviewer: I don't think we should document this on this page:
- It introduces a task to a page that's really meant to be more of a conceptual overview. 
- It kind of separates the 'before' and 'after' steps for manual migration into 2 pages, which I think might be a bit confusing for users. They'd have to come here to prepare for the migration, and then the module installation page for the next steps. 

I'm torn on where to document this stuff, honestly:
- We could add it to the existing migration stuff on the project page, but it seems weird to have a lot of edge-casey module stuff on there.
- We could add a temporary page for manual migration + module management, and link to it from all of the likely places in the docs. 
- We could add it to the "preparing for 3.0 page," but that presents more complications because we already talk about a different kind of manual migration there. 
- We could add it to the installing modules page under a "migrating" section. 

It's a tough one. I think I'm leaning towards just adding a section to the installing modules page at this point. I guess it doesn't have to be perfect because it's temporary, right? Suggestions welcome.

> Existing Bolt projects can be easily updated to use the module management
> feature. In most cases, Bolt can update your project to use this feature
> for you. However, project's with more complex module requirements may
> require manual updating.

Reviewer: ```suggestion
If you created a Bolt project before module dependency management was 
introduced, you can use Bolt's built-in migration tool to [update your project]
(projects.md#migrate-a-bolt-project).

There are a few cases where Bolt is unable to resolve module dependencies and
manage your project's modules for you. If Bolt can't resolve your module 
dependencies, you must manage your project's Puppetfile manually and use Bolt 
to install the modules listed in the Puppetfile without resolving dependencies.

You need to manage your modules manually if:
```

> If your project requires any of the following, you should update your
> project manually:

Reviewer: ```suggestion
```

- Modules installed through a proxy
- Modules installed from a Forge other than the [Puppet
  Forge](https://forge.puppet.com)

> - Modules installed from a git repository other than a public GitHub repository

Reviewer: ```suggestion
- Modules installed from a git repository other than a public GitHub repository

For more information on manually managing your modules, see [Manually manage a project's modules](bolt_installing_modules.md#manually-manage-a-projects-modules).
```

> ### Automated migration
> 
> For information on updating your project using Bolt, see [Migrate a Bolt
> project](projects.md#migrate-a-bolt-project).

Reviewer: ```suggestion
```

> ### Manual migration

Reviewer: Let's move this section to the installing modules page under the "Manually manage a project's modules" section. I think we can treat it as a prerequisite to manually managing modules.
